pageserver/src/tenant/timeline.rs

```diff
@@ -1135,6 +1158,162 @@ impl Timeline {
         );
     }
 
+    fn launch_eviction_task(self: &Arc<Self>) {
+        let ctx = RequestContext::todo_child(TaskKind::Eviction, DownloadBehavior::Error);
+        let self_clone = Arc::clone(self);
+        task_mgr::spawn(
+            BACKGROUND_RUNTIME.handle(),
+            TaskKind::Eviction,
+            Some(self.tenant_id),
+            Some(self.timeline_id),
+            &format!("eviction for {}/{}", self.tenant_id, self.timeline_id),
+            false,
+            async move {
+                self_clone.eviction_task(&ctx).await;
+                Ok(())
+            },
+        );
+    }
+
+    #[instrument(skip_all, fields(tenant_id = %self.tenant_id, timeline_id = %self.timeline_id))]
+    async fn eviction_task(self: Arc<Self>, _ctx: &RequestContext) {
+        loop {
+            let policy = self.get_eviction_policy();
+            debug!("eviction policy is {policy:?}");
+            let sleep_until = match policy {
+                EvictionPolicy::NoEviction => {
+                    // check again in 10 seconds; XXX config watch mechanism
+                    tokio::select! {
+                        _ = tokio::time::sleep(Duration::from_secs(10)) => {},
+                        _ = task_mgr::shutdown_watcher() => {},
+                    }
+                    continue;
+                }
+                EvictionPolicy::LayerAccessThreshold(p) => {
+                    let start = Instant::now();
+                    self.eviction_iteration_threshold(p).await;
+                    let elapsed = start.elapsed();
+                    if elapsed > p.period {
+                        warn!(
+                            configured_period = %humantime::format_duration(p.period),
+                            last_period = %humantime::format_duration(elapsed),
+                            "this eviction period took longer than the configured period"
+                        );
+                    }
+                    start + p.period
+                }
+            };
+            tokio::select! {
+                // XXX replace with context cancellation once that's done
+                _ = task_mgr::shutdown_watcher() => {
+                    info!("shutting down");
+                    return;
+                }
+                _ = tokio::time::sleep_until(sleep_until.into()) => { }
+            }
+        }
+    }
+
+    async fn eviction_iteration_threshold(self: &Arc<Self>, p: EvictionPolicyLayerAccessThreshold) {
+        let now = SystemTime::now();
+        let remote_client = match self.remote_client.as_ref() {
+            None => {
+                error!("no remote storage configured but eviction requires that");
+                // check again in 1 hour; XXX config watch mechanism
```

Comment:
Restart is the only way to reconfigure pageserver.

Reply:
With this I meant that perhaps just error out quickly, or `INFO` out quickly (for tests, local usage)

Reply:
Though ... I guess the functionality without a remote client could be to just "maintain the set of would-be-evicted".